# STAT5342 Project 1 
**By Yiming Zhu yz4336**

# Introduction of dataset
For this project, I use the History of Philosophy dataset. This data site is about the sentences given by different philosophers from different schools. In this dataset, we can exam the pattern, the word choice of differet philosopher. 

Source: https://www.kaggle.com/datasets/kouroshalizadeh/history-of-philosophy?resource=download

#Question1: Among *analytic* and *aristotle*, philosophers from which school prefer long sentences

I wonder if the different teaching methods in different schools will affect the language models of philosophers. For example, some colleges choose textbooks, so philosophers in that college tend to choose longer philosophical sentences. To investigate this question, I choose two schools that I am interested most, analytic and aristotle. 

#Method
To investigate this question, I choose two schools that I am interested most, analytic and aristotle. Firstly, I wrangle the data so that it has two column: *school* and *sentence_length*. Here, school only contains records of analytic and aristotle. 

Then I calculted *Pearson's chi-squared test* of the sentence length across schools to explore whether there were significant differences. Pearson's chi-squared test is a statistical test used to determine if there is a significant difference between the observed and expected frequencies in one or more categorical variables. It is commonly used to determine if there is a relationship between two categorical variables. The test calculates the chi-squared statistic, which is a measure of the difference between the observed and expected frequencies, and compares it to a critical value from the chi-squared distribution to determine if the difference is significant. The result of the test can be used to accept or reject the null hypothesis that the variables are independent.


In [2]:
from google.colab import drive
drive.mount('/content/drive') 
import pandas as pd

philosophy=pd.read_csv('/content/drive/MyDrive/Colab Notebooks/STAT5243/philosophy_data.csv')

Mounted at /content/drive


In [3]:
df=pd.read_csv('/content/drive/MyDrive/Colab Notebooks/STAT5243/philosophy_data.csv')
df.head(10)

title author school  \
0  Plato - Complete Works  Plato  plato   
1  Plato - Complete Works  Plato  plato   
2  Plato - Complete Works  Plato  plato   
3  Plato - Complete Works  Plato  plato   
4  Plato - Complete Works  Plato  plato   
5  Plato - Complete Works  Plato  plato   
6  Plato - Complete Works  Plato  plato   
7  Plato - Complete Works  Plato  plato   
8  Plato - Complete Works  Plato  plato   
9  Plato - Complete Works  Plato  plato   

                                      sentence_spacy  \
0   What's new, Socrates, to make you leave your ...   
1  Surely you are not prosecuting anyone before t...   
2  The Athenians do not call this a prosecution b...   
3                              What is this you say?   
4  Someone must have indicted you, for you are no...   
5                 But someone else has indicted you?   
6        I do not really know him myself, Euthyphro.   
7                He is apparently young and unknown.   
8                  They call him Meletus, I believe.   
9  He belongs to the Pitthean deme, if you know a...   

                                        sentence_str  \
0   What's new, Socrates, to make you leave your ...   
1  Surely you are not prosecuting anyone before t...   
2  The Athenians do not call this a prosecution b...   
3                              What is this you say?   
4  Someone must have indicted you, for you are no...   
5                 But someone else has indicted you?   
6        I do not really know him myself, Euthyphro.   
7                He is apparently young and unknown.   
8                  They call him Meletus, I believe.   
9  He belongs to the Pitthean deme, if you know a...   

   original_publication_date  corpus_edition_date  sentence_length  \
0                       -350                 1997              125   
1                       -350                 1997               69   
2                       -350                 1997               74   
3                       -350                 1997               21   
4                       -350                 1997              101   
5                       -350                 1997               34   
6                       -350                 1997               43   
7                       -350                 1997               35   
8                       -350                 1997               33   
9                       -350                 1997              147   

                                    sentence_lowered  \
0   what's new, socrates, to make you leave your ...   
1  surely you are not prosecuting anyone before t...   
2  the athenians do not call this a prosecution b...   
3                              what is this you say?   
4  someone must have indicted you, for you are no...   
5                 but someone else has indicted you?   
6        i do not really know him myself, euthyphro.   
7                he is apparently young and unknown.   
8                  they call him meletus, i believe.   
9  he belongs to the pitthean deme, if you know a...   

                                       tokenized_txt  \
0  ['what', 'new', 'socrates', 'to', 'make', 'you...   
1  ['surely', 'you', 'are', 'not', 'prosecuting',...   
2  ['the', 'athenians', 'do', 'not', 'call', 'thi...   
3               ['what', 'is', 'this', 'you', 'say']   
4  ['someone', 'must', 'have', 'indicted', 'you',...   
5  ['but', 'someone', 'else', 'has', 'indicted', ...   
6  ['do', 'not', 'really', 'know', 'him', 'myself...   
7  ['he', 'is', 'apparently', 'young', 'and', 'un...   
8      ['they', 'call', 'him', 'meletus', 'believe']   
9  ['he', 'belongs', 'to', 'the', 'pitthean', 'de...   

                                      lemmatized_str  
0     what be new , Socrates , to make -PRON- lea...  
1   surely -PRON- be not prosecute anyone before ...  
2   the Athenians do not call this a prosecution ...  
3                          what be this -PRON- say ?  
4   someone must have indict -PR

In [4]:
df2=df[df['school'].isin(['analytic','aristotle'])].reset_index() 
df3=df2[['school','sentence_length']]

In [5]:
from scipy.stats import chisquare
chisquare(df3['school'].value_counts())

Power_divergenceResult(statistic=423.8735173313884, pvalue=3.5021803536319334e-94)

In our analysis, the p-value of the chi-squared test is less than 0.05. Hence, there is no difference of the sentence length between *analytic* and *aristotle*. 

#Question2: Which philosopher's sentence has the longest average sentence length

I'm curious whether different philosophers have significant differences in the length of sentences. 

#Method
To explore this question, I group the data by the author and calculate the average sentence length for each philosopher and compared them.




In [12]:
df5 = df.groupby('author')['sentence_length'].aggregate(['mean'])
df5

mean
author                     
Aristotle        153.224953
Beauvoir         148.790351
Berkeley         139.653987
Davis            139.671134
Deleuze          163.671850
Derrida          143.431239
Descartes        247.381625
Epictetus        118.430341
Fichte           151.964582
Foucault         189.637467
Hegel            175.720088
Heidegger        118.541965
Hume             180.192372
Husserl          185.473703
Kant             198.159400
Keynes           196.654060
Kripke           119.025082
Leibniz          157.085140
Lenin            181.423137
Lewis            109.717607
Locke            200.395836
Malebranche      164.434023
Marcus Aurelius  139.776221
Marx             143.253466
Merleau-Ponty    170.934009
Moore            167.254907
Nietzsche        116.599867
Plato            114.938018
Popper           139.545105
Quine            121.643429
Ricardo          186.252751
Russell          146.296669
Smith            185.277944
Spinoza          146.544424
Wittgenstein      84.883772
Wollstonecraft   190.957796

Based on the DataFrame above, we can see that different authors have different avgerage sentence length. Wittgenstein has average sentence length of 84.883772 words, while Descartes	has average sentence length of 247.381625. So different authors have very different avgerage sentence length. 